server/src/main/java/org/opensearch/action/support/ActionFilter.java

```diff
@@ -52,14 +52,38 @@ public interface ActionFilter {
     /**
      * Enables filtering the execution of an action on the request side, either by sending a response through the
      * {@link ActionListener} or by continuing the execution through the given {@link ActionFilterChain chain}
+     *
+     * Note: This has only for a transitional period a default implementation. After the deprecated apply()
+     * method has been removed, the default implementation will be removed.
      */
-    <Request extends ActionRequest, Response extends ActionResponse> void apply(
+    default <Request extends ActionRequest, Response extends ActionResponse> void apply(
         Task task,
         String action,
         Request request,
+        ActionRequestMetadata<Request, Response> actionRequestMetadata,
         ActionListener<Response> listener,
         ActionFilterChain<Request, Response> chain
-    );
+    ) {
+        this.apply(task, action, request, listener, chain);
+    }
+
+    /**
+     * Enables filtering the execution of an action on the request side, either by sending a response through the
+     * {@link ActionListener} or by continuing the execution through the given {@link ActionFilterChain chain}
+     *
+     * @deprecated please do not override this method any more. Instead, override the method with the additional
+     * ActionRequestMetadata parameter. This method will be removed soon.
+     */
+    @Deprecated
+    default <Request extends ActionRequest, Response extends ActionResponse> void apply(
+        Task task,
+        String action,
+        Request request,
+        ActionListener<Response> listener,
+        ActionFilterChain<Request, Response> chain
+    ) {
+        assert false : "Incomplete implementation of ActionFilter interface in " + this;
```

Comment:
I think it's better to unconditionally throw an UnsupportedOperationException here. Asserts may or may not be enabled at runtime, which means this might silently be a noop.

Reply:
ok